ament_tools/build_types/ament_cmake.py

```diff
@@ -93,8 +93,7 @@ def on_build(self, context):
         # Calculate any extra cmake args which are not common between cmake build types
         extra_cmake_args = []
         if should_run_configure:
-            if context.build_tests:
-                extra_cmake_args += ["-DAMENT_ENABLE_TESTING=1"]
+            extra_cmake_args += ["-DBUILD_TESTING=%d" % int(context.build_tests)]
```

Comment:
The problem is that before we didn't know when it was intentional and when it was not. I think the new behavior is more consistent - you get what you specify.